*/releases, TOC: publish 3.0 GA release notes (#1289)
pingcap/docs: dev/releases/3.0.0-ga.md

---

> title: TiDB 3.0.0 GA Release Notes

Author: ```suggestion
title: TiDB 3.0 GA Release Notes
```

category: Releases
---

> # TiDB 3.0.0 GA Release Notes

Author: ```suggestion
# TiDB 3.0 GA Release Notes
```

Release date: June 28, 2019

TiDB version: 3.0.0

TiDB Ansible version: 3.0.0

## Overview

> On June 28, 2019, TiDB 3.0.0 GA is released. The corresponding TiDB Ansible version is 3.0.0. Compared with TiDB 2.1, this release has greatly improved in the following aspects:

Author: ```suggestion
On June 28, 2019, TiDB 3.0 GA is released. The corresponding TiDB Ansible version is 3.0.0. Compared with TiDB 2.1, this release has greatly improved in the following aspects:
```

- Stability. TiDB 3.0 has demonstrated long-term stability for large-scale clusters with up to 150+ nodes and 300+TB of storage.
- Usability. TiDB 3.0 has multi-facet improvements in usability, including standardized slow query logs, well-developed log file specification, and new features such as `EXPLAIN ANALYZE` and SQL Trace to save operation costs for users.

> - Performance. TiDB 3.0 has approximately 3.5 times of performance improvements with TPC-C and 1.5 times with Sysbench. Thanks to the implementation of Views, TPC-H 50G Q5 and Q15 have both yielded normal results.

Author: ```suggestion
- Performance. TiDB 3.0 has approximately 4.5 times of performance improvements with TPC-C and 1.5 times with Sysbench. Thanks to the implementation of Views, TPC-H 50G Q5 and Q15 have both yielded normal results.
```

- New features including Window Functions, Views (**Experimental**), partitioned tables, the plugin framework, pessimistic locking (**Experimental**), and `SQL Plan Management`.

## TiDB

+ New Features

>     - Support Window Functions; compatible with all window functions in MySQL 8.0, including `NTILE`, `LEAD`, `LAG`, `PERCENT_RANK`, `NTH_VALUE`, `CUME_DIST`,      `FIRST_VALUE` , `LAST_VALUE`, `RANK`, `DENSE_RANK`, and `ROW_NUMBER`

Reviewer: ```suggestion
    - Support Window Functions; compatible with all window functions in MySQL 8.0, including `NTILE`, `LEAD`, `LAG`, `PERCENT_RANK`, `NTH_VALUE`, `CUME_DIST`, `FIRST_VALUE` , `LAST_VALUE`, `RANK`, `DENSE_RANK`, and `ROW_NUMBER`
```

        - Support Range Partition
        - Support Hash Partition
    - Add the plug-in framework, supporting plugins such as IP Whitelist (**Enterprise**) and Audit Log (**Enterprise**).
    - Support the SQL Plan Management function to create SQL execution plan binding to ensure query stability (**Experimental**)
+ SQL Optimizer
    - Optimize the `NOT EXISTS` subquery and convert it to `Anti Semi Join` to improve performance
    - Optimize the constant propagation on the `Outer Join`, and add the optimization rule of `Outer Join` elimination to reduce non-effective computations and improve performance
    - Optimize the `IN` subquery to execute `Inner Join` after aggregation to improve performance
    - Optimize `Index Join` to adapt to more scenarios
    - Improve the Partition Pruning optimization rule of Range Partition
    - Optimize the query logic for `_tidb_rowid`to avoid full table scan and improve performance
    - Match more prefix columns of the indexes when extracting access conditions of composite indexes if there are relevant columns in the filter to improve performance
    - Improve the accuracy of cost estimates by using order correlation between columns

>     - Optimize `Join Reorder` based on the Greedy algorithm and the dynamic planning algorithm to improve accuracy for index selection using `Join`

Author: ```suggestion
    - Optimize `Join Reorder` based on the Greedy algorithm and the dynamic planning algorithm to improve the join order selection when joining multiple tables
```

>     - Optimize `Join Reorder` based on the Greedy algorithm and the dynamic planning algorithm to improve accuracy for index selection using `Join`

Author: ```suggestion
    - Optimize `Join Order` based on the Greedy algorithm and the dynamic planning algorithm to speed up the join operation of multiple tables
```
